R/paren_body_linter.R

```diff
@@ -0,0 +1,33 @@
+#' @describeIn linters check that there is a space between the
+#' right parenthesis and the function body when no braces are used
+#' to define a function.
+#'
+#' @export
+paren_body_linter <- function() {
+  Linter(function(source_file) {
+    if (is.null(source_file$xml_parsed_content)) return(NULL)
+    
+    xpath <- paste(
+      "//expr[",
+      "@line1 = preceding-sibling::FUNCTION/@line1",
+      "and",
+      "@col1 = preceding-sibling::OP-RIGHT-PAREN/@col1 + 1",
+      "]"
+    )
+    matched_expressions <- xml2::xml_find_all(source_file$xml_parsed_content, xpath)
+    
+    lapply(matched_expressions, get_lint_from_expression, source_file = source_file)
+  })
+}
+
+get_lint_from_expression <- function(expression, source_file) {
```

Comment:
This looks very much like an alternative implementation of `xml_nodes_to_lint` in `R/utils.R`:
https://github.com/jimhester/lintr/blob/81e3b466eec949ef340e43add3207b2cca9ff86b/R/utils.R#L202-L226

Can you check if you can use that function instead?

Reply:
Refactored to xml_nodes_to_lint. It flew under my radar. Not all xml-based linters use this (e.g. paren_brace_linter).